pkg/conditions/upgrading.go

```diff
@@ -28,15 +21,21 @@ func IsUpgradingInProgress(cr CR, desiredRelease string) (bool, error) {
 		return false, nil
 	}
 
-	msg, err := aeconditions.DeserializeUpgradingConditionMessage(c.Message)
-	if err != nil {
-		return false, microerror.Mask(err)
+	// release.giantswarm.io/version is updated, which means that the node pool
+	// upgrade has been started, let's see if it is still in progress.
+
+	lastDeployedReleaseVersion, isSet := cr.GetAnnotations()[annotation.LastDeployedReleaseVersion]
+	if !isSet {
+		// If release.giantswarm.io/last-deployed-version is not set, that
+		// means that the node pool is still being created, so no upgrade in
+		// progress.
+		return false, nil
 	}
 
-	// If release version in condition metadata contains desired release
-	// version while condition status is not True, it means that upgrade has
+	// If last deployed release version is equal to the desired release version
+	// while Upgrading condition status is not True, it means that upgrade has
 	// been completed already.
-	if msg.ReleaseVersion == desiredRelease {
+	if lastDeployedReleaseVersion == desiredRelease {
```

Comment:
@tuommaki Now we just check the annotation here, if the matches the desired release, that means that the upgrade has been completed for the specified desired release version.